src/test/kotlin/uk/gov/dluhc/printapi/service/aed/AnonymousElectorDocumentSearchServiceTest.kt

```diff
@@ -38,141 +38,50 @@ internal class AnonymousElectorDocumentSearchServiceTest {
     @Mock
     private lateinit var anonymousSearchSummaryMapper: AnonymousSearchSummaryMapper
 
+    @Mock
+    private lateinit var specificationBuilder: AnonymousElectorDocumentSummarySpecificationBuilder
+
     @InjectMocks
     private lateinit var anonymousElectorDocumentSearchService: AnonymousElectorDocumentSearchService
 
     @Test
-    fun `should return empty Anonymous Elector Document summaries given null searchBy`() {
```

Comment:
I've removed all but one of the previous unit test methods in this class (`AnonymousElectorDocumentSearchServiceTest`)

The implementation has changed, so clearly I needed to make some change to this test. Now that the implementation is using the specification builder it actually becomes hard to test. The key part of testing the implementation is the `Specification` which is built by the SpecificationBuilder.
I could setup mocks to return a `Specification` in various states to reflect querying by surname/appRef/nothing etc etc .... but in the context of unit testing the service it adds nothing

So I chose to have 1 simple test that basically just asserts that the various collaborators are called with the correct/expected values .... and then leave testing of the actual logic wrt returned values etc to the new `AnonymousElectorDocumentSearchServiceIntegrationTest` above ☝️